src/bootstrap/src/core/config/config.rs

```diff
@@ -457,6 +457,8 @@ impl std::str::FromStr for RustcLto {
 }
 
 #[derive(Copy, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+// N.B.: This type is used everywhere, and the entire codebase relies on it being Copy.
+// Making !Copy is highly nontrivial!
 pub struct TargetSelection {
     pub triple: Interned<String>,
     file: Option<Interned<String>>,
```

Comment:
Maybe use `&'static str` here by leaking the strings using `Box::leak`? There should only be a handful of unique `TargetSelection`s anyway.